### PR TITLE
[Fix] Fix some loading bugs and support fov_image_based mode in Waymo dataset. 

### PR DESCRIPTION
**Pull request by lianqing11**
## Motivation

1. Utilize load_type to distinguish different training modes in dataset loading and evaluation. 
2. Fix the bug of waymo_converter when handle testing_camera_only split
3. Add print information during loading ann file.

### CODE REVIEW

File: mmdet3d/datasets/nuscenes_dataset.py [hunk 1]
```diff
@@ -33,6 +32,14 @@ class NuScenesDataset(Det3DDataset):
             - 'LiDAR': Box in LiDAR coordinates.
             - 'Depth': Box in depth coordinates, usually for indoor dataset.
             - 'Camera': Box in camera coordinates.
+        load_type (str, optional): Type of loading mode.
```
Comment (JingweiZhang12) [inline mmdet3d/datasets/nuscenes_dataset.py:35]: Please add `Defaults to xxx. and check all the other similar docstrings.

**Comment by JingweiZhang12** [comment]
Please fix the UT and add some description about the mapping relation between the values of the old argument  'task' and the new argument 'load_type'.

**Review by lianqing11 — COMMENTED** [review]
should be sample_idx

File: mmdet3d/datasets/kitti_dataset.py [hunk 2]
```diff
@@ -32,6 +32,14 @@ class KittiDataset(Det3DDataset):
             - 'LiDAR': Box in LiDAR coordinates.
             - 'Depth': Box in depth coordinates, usually for indoor dataset.
             - 'Camera': Box in camera coordinates.
+        load_type (str, optional): Type of loading mode.
```
Comment (ZCMax) [inline mmdet3d/datasets/kitti_dataset.py:35]: It seems the `load_type` design conflicts with the unified `task` design.

File: mmdet3d/datasets/kitti_dataset.py [hunk 3]
```diff
@@ -32,6 +32,14 @@ class KittiDataset(Det3DDataset):
...
+            - 'frame_based': Load all of the instances in the frame.
+            - 'mv_image_based': Load all of the instances in the frame and need
+                to convert to the FOV-based data type to support image-based
+                detector.
+            - 'fov_image_base': Only load the instances inside the default cam,
```
Comment (ZCMax) [inline mmdet3d/datasets/kitti_dataset.py:40]: `fov_image_based`

File: mmdet3d/datasets/kitti_dataset.py (hunk 2 above)
Comment (ZCMax) [reply mmdet3d/datasets/kitti_dataset.py:35]: Whether `lidar_det`, `multi-view_det` and `mono_det` can match `frame_based`, `mv_image_based`, `for_image_based`?

Comment (Tai-Wang) [reply mmdet3d/datasets/kitti_dataset.py:35]: Cannot. lidar_det corresponds to frame_based and mono_det corresponds to fov_image_based exactly, while multi-view_det may correspond to frame_based/mv_image_based.

**Review by Tai-Wang — APPROVED** [review]

**Pull request merged**